service/worker/batcher/activities.go

```diff
@@ -41,6 +42,258 @@ var (
 	errNamespaceMismatch = errors.New("namespace mismatch")
 )
 
+// batchProcessorConfig holds the configuration for batch processing
+type batchProcessorConfig struct {
+	namespace         string
+	adjustedQuery     string
+	rps               float64
+	concurrency       int
+	initialPageToken  []byte
+	initialExecutions []*commonpb.WorkflowExecution
+}
+
+// batchWorkerProcessor defines the interface for different worker processor types
+type batchWorkerProcessor func(
+	ctx context.Context,
+	taskCh chan taskDetail,
+	respCh chan taskResponse,
+	rateLimiter *rate.Limiter,
+	sdkClient sdkclient.Client,
+	frontendClient workflowservice.WorkflowServiceClient,
+	metricsHandler metrics.Handler,
+	logger log.Logger,
+)
+
+// Task represents a task to be processed (alias for taskDetail for consistency with pseudocode)
+type Task = taskDetail
+
+// Page represents a page of workflow executions to be processed
+type Page struct {
```

Comment:
I know I suggested this name but I realized we shouldn't export it:

```suggestion
type page struct {
```